common/utils/src/main/resources/error/README.md

```diff
@@ -96,21 +132,21 @@ The quality of the error message should match the
 ### SQLSTATE
 
 SQLSTATE is an mandatory portable error identifier across SQL engines.
-SQLSTATE comprises a 2-character class value followed by a 3-character subclass value.
+SQLSTATE comprises a 2-character category followed by a 3-character sub-category.
```

Comment:
Note the changes in this section per the proposal in [SPARK-46810](https://issues.apache.org/jira/browse/SPARK-46810).

Reply:
class and subclass are the terms used in the standard. But I'm not married to them....

Reply:
Oh. So the SQL standard calls `42` an "error class", and also calls `INCOMPLETE_TYPE_DEFINITION` an "error class"?

Reply:
@nchammas The relevant section is ISO/IEC 9075-2:2016(E) 24.1 SQLSTATE
The character string value returned in an SQLSTATE parameter comprises a 2-character class code followed by a 3-character subclass code, each with an implementation-defined character set that has a one-octet character encoding form and is restricted to <digit>s and <simple Latin upper case letter>s. Table 38, “SQLSTATE class and subclass codes”, specifies the class code for each condition and the subclass code or codes for each class code.
Class codes that begin with one of the <digit>s '0', '1', '2', '3', or '4' or one of the <simple Latin upper case letter>s 'A', 'B', 'C', 'D', 'E', 'F', 'G', or 'H' are returned only for conditions defined in ISO/IEC 9075 or in any other International Standard. The range of such class codes is called standard-defined classes. Some such class codes are reserved for use by specific International Standards, as specified elsewhere in this Clause. Subclass codes associated with such classes that also begin with one of those 13 characters are returned only for conditions defined in ISO/IEC 9075 or some other International Standard. The range of such subclass codes is called standard-defined subclasses. Subclass codes associated with such classes that begin with one of the <digit>s '5', '6', '7', '8', or '9' or one of the <simple Latin upper case letter>s 'I', 'J', 'K', 'L', 'M', 'N', 'O', 'P', 'Q', 'R', 'S', 'T', 'U', 'V', 'W', 'X', 'Y', or 'Z' are reserved for implementation-defined conditions and are called implementation- defined subclasses.
Class codes that begin with one of the <digit>s '5', '6', '7', '8', or '9' or one of the <simple Latin upper case letter>s 'I', 'J', 'K', 'L', 'M', 'N', 'O', 'P', 'Q', 'R', 'S', 'T', 'U', 'V', 'W', 'X', 'Y', or 'Z' are reserved for implementation-defined exception conditions and are called implementation-defined classes. All subclass codes except '000', which means no subclass, associated with such classes are reserved for implementation-defined conditions and are called implementation-defined subclasses. An implementation-defined completion condition shall be indicated by returning an implementation-defined subclass in conjunction with one of the classes successful completion, warning, or no data.